contrib/IECoreAlembic/src/IECoreAlembic/AlembicScene.cpp

```diff
@@ -267,6 +319,493 @@ class AlembicScene::AlembicReader : public AlembicIO
 			}
 		}
 
+		// Attribute
+		// =--------
+
+		bool hasAttribute( const Name &name ) const
+		{
+			if( !m_xform )
+			{
+				return false;
+			}
+
+			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
+
+			if( !userProperties.valid() )
+			{
+				return false;
+			}
+
+			const AbcA::PropertyHeader *propertyHeader = userProperties.getPropertyHeader( name.string() );
+
+			return propertyHeader != nullptr;
+		}
+
+		void attributeNames( NameList &attrs ) const
+		{
+			attrs.clear();
+
+			if( !m_xform )
+			{
+				return;
+			}
+
+			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
+
+			if( !userProperties.valid() )
+			{
+				return;
+			}
+
+			attrs.reserve( userProperties.getNumProperties() );
+
+			for( size_t i = 0; i < userProperties.getNumProperties(); ++i )
+			{
+				attrs.push_back( InternedString( userProperties.getPropertyHeader( i ).getName() ) );
+			}
+		}
+
+		size_t numAttributeSamples( const Name &name ) const
+		{
+			if( !m_xform )
+			{
+				return 0;
+			}
+
+			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
+
+			const PropertyHeader *header = userProperties.getPropertyHeader( name.string() );
+
+			if( !header )
+			{
+				return 0;
+			}
+
+			return header->getTimeSampling()->getNumStoredTimes();
+		}
+
+		double attributeSampleTime( const Name &name, size_t sampleIndex ) const
+		{
+			if( !m_xform )
+			{
+				return 0.0;
+			}
+
+			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
+			double t = userProperties.getPropertyHeader( name.string() )->getTimeSampling()->getSampleTime( sampleIndex );
+			return t;
+		}
+
+		double attributeSampleInterval( const Name &name, double time, size_t &floorIndex, size_t &ceilIndex ) const
+		{
+			if( !m_xform )
+			{
+				return 0.0;
+			}
+
+			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
+			const PropertyHeader *propertyHeader = userProperties.getPropertyHeader( name.string() );
+
+			if( !propertyHeader )
+			{
+				return 0.0;
+			}
+
+			auto timeSampling = propertyHeader->getTimeSampling();
+
+			return sampleInterval( timeSampling.get(), timeSampling->getNumStoredTimes(), time, floorIndex, ceilIndex );
```

Comment:
Again here, the number of samples needs to come frm the property, not `timeSampling->getNumStoredTimes()`.